libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -363,10 +362,27 @@ void ContractLevelChecker::checkBaseConstructorArguments(ContractDefinition cons
 	// check that we get arguments for all base constructors that need it.
 	// If not mark the contract as abstract (not fully implemented)
 	for (ContractDefinition const* contract: bases)
-		if (FunctionDefinition const* constructor = contract->constructor())
-			if (contract != &_contract && !constructor->parameters().empty())
-				if (!_contract.annotation().baseConstructorArguments.count(constructor))
-					_contract.annotation().unimplementedDeclarations->push_back(constructor);
+		if (FunctionDefinition const* baseConstructor = contract->constructor())
+			if (
+				_contract.contractKind() == ContractKind::Contract &&
```

Comment:
We could pull the first two conditions outside the loop, it makes it easier to read (since `contract` is not yet in scope) and also avoids the loop.

Reply:
> Although we could also remove the first condition - it should be true anyway.

I tried to replace it with an assert but looks like it's actually needed. `_contract` could be a library or an interface. Making it an assert makes multiple tests fail, including ones as simple as [`interface_virtual_warning.sol`](https://github.com/ethereum/solidity/blob/develop/test/libsolidity/syntaxTests/inheritance/interface_virtual_warning.sol).

Reply:
If `contract` and `_contract` are different, then it has to be a contract, because libraries and intrefaces cannot inherit